### ISSUE TITLE
Popp 701721 TRV gets firmware offered via OTA which is "not officially released" by vendor, and which supersedes newer device firmware

### ISSUE THREAD

**Issue by ernstkl** [Koenkk/zigbee2mqtt#15122]
### What happened?

Ca. beginning of 2022-Nov there was a firmware update "01.18.0008 01.18" offered for OTA.
I did the update. 

Later when I contacted the vendor (Popp) for reasons not relevant here they told me "the firmware your TRV runs is a Danfoss firmware and it was never approved by us".

Even more serious is the observation that this OTA firmware overwrites a more up-to date device firmware.
A new 701271 I own runs a firmware dated 20220524 with version "01.12.0008 01.12."

The firmware "01.18.0008 01.18" offered for OTA has a much earlier release date: 20210222

### What did you expect to happen?

1) Maybe a Danfoss firmware should not be offered for Popp devices because this results in issues with vendor support?

2) For for deciding whether a firmware is offered for a specific device maybe both date _and_ firmware version numbers should be compared.

### Zigbee2MQTT version

1.28.2 

### Adapter firmware version

20220219

### Adapter

SONOFF Zigbee 3.0 USB Dongle Plus (ZBDongle-P)

**Comment by rsporsche** [comment]
"pi_heating_demand often gets stuck at 1%, even when setpoint is 20 deg below current temperature for hours"

I have this same problem, I thought that the POPP was essentially identical to the Danfoss though so I can't see why it would behave differently. If POPP aren't supporting their devices it might be worth paying the little extra for Danfoss.

**Comment by ernstkl** [comment]
Meanwhile I experienced the "stuck at 1%" problem also with the new device on which no OTA has been performed. So this problem is not related to the "01.18.0008 01.18" Danfoss firmware.

**Comment by ernstkl** [comment]
> Meanwhile I experienced the "stuck at 1%" problem also with the new device on which no OTA has been performed. So this problem is not related to the "01.18.0008 01.18" Danfoss firmware.

So as the "stuck at 1%" issue is not related to the "wrong firmware installed" issue, it is off topic.
For that reason I am now going to edit the issue title and description to make it clean and to the point.

Continuing the "off topic" discussion shortly for the possible benefit of readers: I got a reply by Popp today reg. the 1%-issue. They tell me (summary): 
> "A valve that is OK closes already at about 10% actuator position. The remaining travel down to zero normally just results in compressing the rubber seal. Closing down to 1% is what the TRV does for a valve that is detected to be problematic." 

Apparently, in my case this was still not enough to stop the heating water, so I will have to fix that.
So "stuck at 1%" is not an error but by design.

**Comment by gebauer** [comment]
Hi,

this is an eye-opener for me.

I have 5 POP devices, three of them I bought only recently. They are on the 01.12.0008 01.12 firmware from 20220929 (which I didn't notice). 
I now tried for several hours to update them to 01.18.0008 01.18, which seems to be an Danfoss firmware, which is older.
Luckily (?) the devices always aborted the update after 1-2h.

It is very confusing the Danfoss and POP have different firmwares with the same firmware stye - but with different meanings.

The 01.12 from Danfoss is pretty old and does not support externals temeperature sensors very good. The 0.12 from POP seems to very modern and exposes everything the 1.18 from Danfos does...
Somehow Zigbee2MQTT has to handle with this 🤷‍♂️

Best
Jan

**Comment by ernstkl** [comment]
I do not have any firmware file.

**Comment by wintersteiger** [comment]
I arrived here because I'm looking for updated firmware for a Popp valve (currently at 01.12.0008 01.12 dated 20220927, but not currently using zigbee2mqtt). I noticed that my valve reports a manufacturer ID of 1246, which is Danfoss' ID, i.e., Popp really should be using their own ID if they don't "approve" of Danfoss' firmware.

**Comment by B0rax** [comment]
Was there any solution? Is there a way to update the Popp TRVs through Z2M or not?

**Comment by k3a** [comment]
I updated my Pop using Z2M to the Denfoss 01.18.0008 01.18 firmware, and it appears to be working properly.
Initially, I noticed a strange behavior in which TRV would freeze.
I assumed it was because of the unofficial firmware, but it also happened to one TRV that I did not upgrade.

However, all TRVs have been working perfectly since then.

So upgrading appears to be safe, but there is no way to downgrade. I recommend not upgrading unless you have a specific problem that you know should be fixed in a new firmware.
